src/goto-programs/mm_io.cpp

```diff
@@ -105,26 +106,37 @@ void mm_io(
   }
 }
 
-void mm_io(
-  const symbol_tablet &symbol_table,
-  goto_functionst &goto_functions)
+void mm_io(symbol_tablet &symbol_table, goto_functionst &goto_functions)
 {
   const namespacet ns(symbol_table);
-  exprt mm_io_r=nil_exprt(), mm_io_w=nil_exprt();
+  exprt mm_io_r = nil_exprt(), mm_io_r_value = nil_exprt(),
+        mm_io_w = nil_exprt();
 
   irep_idt id_r=CPROVER_PREFIX "mm_io_r";
   irep_idt id_w=CPROVER_PREFIX "mm_io_w";
 
   auto maybe_symbol=symbol_table.lookup(id_r);
   if(maybe_symbol)
+  {
     mm_io_r=maybe_symbol->symbol_expr();
 
+    const auto &value_symbol = get_fresh_aux_symbol(
+      to_code_type(mm_io_r.type()).return_type(),
+      id2string(id_r) + "$value",
+      id2string(id_r) + "$value",
+      maybe_symbol->location,
+      maybe_symbol->mode,
+      symbol_table);
+
+    mm_io_r_value = value_symbol.symbol_expr();
+  }
+
```

Comment:
Why are the changes to `mm_io` necessary?

Reply:
The current implementation assumes that the return value is in the function#return_value variable. It needs to be changed to use the lhs in the function call.